hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/balancer/TestContainerBalancer.java

```diff
@@ -131,28 +190,255 @@ public void unBalancedNodesListShouldBeEmptyWhenClusterIsBalanced() {
     balancerConfiguration.setThreshold(0.99);
     containerBalancer.start(balancerConfiguration);
 
-    Assert.assertEquals(0, containerBalancer.getUnBalancedNodes().size());
+    // waiting for balance completed.
+    // TODO: this is a temporary implementation for now
+    // modify this after balancer is fully completed
+    try {
+      Thread.sleep(100);
+    } catch (InterruptedException e) {}
+
     containerBalancer.stop();
+    Assert.assertEquals(0, containerBalancer.getUnBalancedNodes().size());
   }
 
   /**
-   * Checks whether ContainerBalancer stops when the limit of
-   * MaxDatanodesToBalance is reached.
+   * ContainerBalancer should not involve more datanodes than the
+   * MaxDatanodesToInvolvePerIteration limit.
    */
   @Test
-  public void containerBalancerShouldStopWhenMaxDatanodesToBalanceIsReached() {
-    balancerConfiguration.setMaxDatanodesToBalance(2);
+  public void containerBalancerShouldObeyMaxDatanodesToInvolveLimit() {
+    balancerConfiguration.setMaxDatanodesToInvolvePerIteration(0.3d);
+    balancerConfiguration.setMaxSizeToMovePerIteration(100 * OzoneConsts.GB);
+    balancerConfiguration.setThreshold(0.01);
+    balancerConfiguration.setIdleIteration(1);
+    containerBalancer.start(balancerConfiguration);
+
+    // waiting for balance completed.
+    // TODO: this is a temporary implementation for now
+    // modify this after balancer is fully completed
+    try {
+      Thread.sleep(1000);
+    } catch (InterruptedException e) {}
+
+    Assert.assertFalse(
```

Comment:
NIT: We could also add assertions for metrics here. Can be done in a separate PR.